backend/inspirehep/disambiguation/tasks.py

```diff
@@ -93,23 +93,51 @@ def match_literature_author_with_config(author_data, matcher_config):
 
 
 def get_reference_and_bai_if_unambiguous_match(matched_refs_ids):
-    if len(matched_refs_ids) == 1:
-        author_reference, author_ids = matched_refs_ids.popitem()
+    if len(matched_refs_ids) != 1:
+        return
+    author_reference, author_ids = matched_refs_ids.popitem()
+    author_bais = get_values_for_schema(author_ids, "INSPIRE BAI")
+    return {
+        "author_reference": author_reference,
+        "author_bai": author_bais[0] if author_bais else None,
+    }
+
+
+def get_reference_and_bai_for_stub_author_if_ambiguous_match(matched_refs_ids):
+    if len(matched_refs_ids) < 1:
+        return
```

Comment:
do you need this? won't the `for` loop afterwards iterate on an empty dict and return `None` implicitly in this case anyway?